### PR TITLE
feat: eslint-plugin-(ember-data|warp-drive)

### PR DESCRIPTION
**Pull request by runspired**
Adds rules to our eslint plugin:

| Impl | Test | Rule | Fix | Recommended | Type Aware |
| ---- | ---- | ---- | --- | ----------- | ---------- |
| ☑️  | ☑️  | no-create-record-rerender    | ❌ | ✅ | 🅧 |
| ☑️  | ☑️  | no-invalid-relationships     | ❌ | ✅ | 🅧 |
| ☑️  | ☑️  | no-legacy-request-patterns   | ❌ | ✅ | 🅧 |
| ☑️  | ☑️  | no-external-request-patterns | ❌ | ✅ | 🅧 |

### CODE REVIEW

File: packages/eslint-plugin-warp-drive/docs/no-create-record-rerender.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+> [!Note]
+> This rule helps applications avoid patterns that often lead to excess or broken renders.
+
+`store.createRecord` creates a record available to
+the application as a whole. For instance: say we create a new `'user'` and add it to the `friends` relationship of `user:2`. When we do this, two distinct
+changes are immediately observable to the entirety of
+the application:
+
+- the list of all users is update (notifying the addition to the LiveArray used by `peekAll` and `findAll`)
```
Comment (gitKrystan) [inline packages/eslint-plugin-warp-drive/docs/no-create-record-rerender.md:15]:
```suggestion
- the list of all users is updated (notifying the addition to the LiveArray used by `peekAll` and `findAll`)
```

File: packages/eslint-plugin-warp-drive/docs/no-invalid-relationships.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# eslint-plugin-warp-drive
+
+| Rule | 🏷️ | ✨ |
+| ---- | -- | -- |
+| `no-invalid-relationships` | 🏆 | ✅ |
+
+> [!TIP]
+> An partial/complete autofix for this rule is possible but has not been implemented.
```
Comment (gitKrystan) [inline packages/eslint-plugin-warp-drive/docs/no-invalid-relationships.md:8]:
```suggestion
> A partial/complete autofix for this rule is possible but has not been implemented.
```

File: packages/eslint-plugin-warp-drive/docs/no-legacy-request-patterns.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# eslint-plugin-warp-drive
+
+| Rule | 🏷️ | ✨ |
+| ---- | -- | -- |
+| `no-legacy-request-patterns` | 🏆 | ✅ |
+
+> [!TIP]
+> An partial autofix for this rule is possible but has not been implemented.
```
Comment (gitKrystan) [inline packages/eslint-plugin-warp-drive/docs/no-legacy-request-patterns.md:8]:
```suggestion
> A partial autofix for this rule is possible but has not been implemented.
```

File: packages/eslint-plugin-warp-drive/src/rules/no-create-record-rerender.js [hunk 4]
```diff
@@ -0,0 +1,164 @@
+// @ts-check
+
+const messageId = 'noCreateRecordRerender';
+const createRecordExpression = 'CallExpression[callee.property.name="createRecord"]';
+const emberObjectExtendExpression = 'CallExpression[callee.property.name="extend"]';
+const forbiddenParentMethodKeyNames = [
+  'init',
+  'initRecord', // legacy modals
```
Comment (gitKrystan) [inline packages/eslint-plugin-warp-drive/src/rules/no-create-record-rerender.js:8]: IIRC this is AB-specific.

Comment (runspired) [reply packages/eslint-plugin-warp-drive/src/rules/no-create-record-rerender.js:8]: yeah, may need to add config to allow additional methods to be disallowed

**Pull request merged**